context-config-store/store/context_config_store_impl.go

```diff
@@ -0,0 +1,139 @@
+			return kurtosisContext, nil
+		}
+	}
+	return nil, stacktrace.NewError("Unable to find current context info in context config file. "+
+		"Current context is set to '%s' but known contexts are: '%s'",
+		currentContextUuid.GetValue(),
+		strings.Join(knownContexts, ", "))
+}
+
+func (store *contextConfigStoreImpl) SwitchContext(contextUuid *api.ContextUuid) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	var doesExist bool
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		doesExist = doesExist || kurtosisContext.GetUuid().GetValue() == contextUuid.Value
+	}
+	if !doesExist {
+		return stacktrace.NewError("Context with UUID '%s' does not exist yet.", contextUuid.GetValue())
+	}
+
+	newContextConfig := api.NewKurtosisContextConfig(contextUuid, contextConfig.GetContexts()...)
+	if err = store.storage.PersistContextConfig(newContextConfig); err != nil {
+		return stacktrace.Propagate(err, "Unable to persist new current context to store")
+	}
+	return nil
+}
+
+func (store *contextConfigStoreImpl) AddNewContext(contextToAdd *api.KurtosisContext) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	newContextUuid := contextToAdd.GetUuid()
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		if kurtosisContext.GetUuid().GetValue() == newContextUuid.GetValue() {
+			return nil // doing nothing as context already exists
+		}
+	}
+
+	var newContexts []*api.KurtosisContext
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		newContexts = append(newContexts, kurtosisContext)
+	}
+	newContexts = append(newContexts, contextToAdd)
+
+	newContextConfig := api.NewKurtosisContextConfig(contextConfig.GetCurrentContext(), newContexts...)
+	if err = store.storage.PersistContextConfig(newContextConfig); err != nil {
+		return stacktrace.Propagate(err, "Unable to persist new context config to store")
+	}
+	return nil
+}
+
+func (store *contextConfigStoreImpl) RemoveContext(contextUuid *api.ContextUuid) error {
+	store.Lock()
+	defer store.Unlock()
+
+	contextConfig, err := store.storage.LoadContextConfig()
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to load current context config")
+	}
+
+	foundContextToRemove := false
+	var newContexts []*api.KurtosisContext
+	for _, kurtosisContext := range contextConfig.GetContexts() {
+		if kurtosisContext.GetUuid().GetValue() != contextUuid.GetValue() {
+			newContexts = append(newContexts, kurtosisContext)
+		} else {
+			foundContextToRemove = true
+		}
+	}
+
+	if !foundContextToRemove {
+		return nil
+	}
+
+	newContextConfig := api.NewKurtosisContextConfig(contextConfig.GetCurrentContext(), newContexts...)
```

Comment:
how does this behave if we remove the current context? should we throw or do we let it get into an inconsistent state?